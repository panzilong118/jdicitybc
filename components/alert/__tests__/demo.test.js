/**
 * @jest-environment node
 */

import * as React from "react";

import renderer from 'react-test-renderer';

import Banner from '../demo/Banner';

import Basic from '../demo/Basic';

import Closable from '../demo/Closable';

import Closetext from '../demo/Closetext';

import Description from '../demo/Description';

import Icon from '../demo/Icon';

import Style from '../demo/Style';

describe('<Alert />', () => {
    it('Banner 顶部公告', () => {
        const demo = renderer.create(<Banner />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Basic 基本用法', () => {
        const demo = renderer.create(<Basic />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Closable 可关闭的警告提示', () => {
        const demo = renderer.create(<Closable />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Closetext 自定义关闭', () => {
        const demo = renderer.create(<Closetext />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Description 含有辅助性文字介绍', () => {
        const demo = renderer.create(<Description />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Icon 图标', () => {
        const demo = renderer.create(<Icon />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Style 四种样式', () => {
        const demo = renderer.create(<Style />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });
});
