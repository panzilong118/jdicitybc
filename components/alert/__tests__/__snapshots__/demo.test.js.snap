// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Banner 顶部公告 1`] = `
<div>
  <div
    className="jc-alert jc-alert-warning jc-alert-banner"
    data-show={true}
    style={undefined}
  >
    <i
      className="jcicon jcicon-exclamation-circle  jc-alert-icon"
      style={Object {}}
    />
    <span
      className="jc-alert-message"
    >
      Warning text
    </span>
    <span
      className="jc-alert-description"
    />
  </div>
  <br />
  <div
    className="jc-alert jc-alert-warning jc-alert-banner"
    data-show={true}
    style={undefined}
  >
    <i
      className="jcicon jcicon-exclamation-circle  jc-alert-icon"
      style={Object {}}
    />
    <span
      className="jc-alert-message"
    >
      Very long warning text warning text text text text text text text
    </span>
    <span
      className="jc-alert-description"
    />
    <a
      className="jc-alert-close-icon"
      onClick={[Function]}
    >
      <i
        className="jcicon jcicon-cross "
        style={Object {}}
      />
    </a>
  </div>
  <br />
  <div
    className="jc-alert jc-alert-warning jc-alert-no-icon jc-alert-banner"
    data-show={true}
    style={undefined}
  >
    <span
      className="jc-alert-message"
    >
      Warning text without icon
    </span>
    <span
      className="jc-alert-description"
    />
  </div>
  <br />
  <div
    className="jc-alert jc-alert-error jc-alert-banner"
    data-show={true}
    style={undefined}
  >
    <i
      className="jcicon jcicon-cross-circle  jc-alert-icon"
      style={Object {}}
    />
    <span
      className="jc-alert-message"
    >
      Error text
    </span>
    <span
      className="jc-alert-description"
    />
  </div>
</div>
`;

exports[`<Alert /> Basic 基本用法 1`] = `
<div
  className="jc-alert jc-alert-success jc-alert-no-icon"
  data-show={true}
  style={undefined}
>
  <span
    className="jc-alert-message"
  >
    Success Text
  </span>
  <span
    className="jc-alert-description"
  />
</div>
`;

exports[`<Alert /> Closable 可关闭的警告提示 1`] = `
<div
  className="jc-alert jc-alert-error jc-alert-with-description jc-alert-no-icon"
  data-show={true}
  style={undefined}
>
  <span
    className="jc-alert-message"
  >
    Error Text
  </span>
  <span
    className="jc-alert-description"
  >
    Error Description Error Description Error Description Error Description Error Description Error Description
  </span>
  <a
    className="jc-alert-close-icon"
    onClick={[Function]}
  >
    <i
      className="jcicon jcicon-cross "
      style={Object {}}
    />
  </a>
</div>
`;

exports[`<Alert /> Closetext 自定义关闭 1`] = `
<div
  className="jc-alert jc-alert-info jc-alert-no-icon"
  data-show={true}
  style={undefined}
>
  <span
    className="jc-alert-message"
  >
    Info Text
  </span>
  <span
    className="jc-alert-description"
  />
  <a
    className="jc-alert-close-icon"
    onClick={[Function]}
  >
    Close Now
  </a>
</div>
`;

exports[`<Alert /> Description 含有辅助性文字介绍 1`] = `
<div
  className="jc-alert jc-alert-error jc-alert-with-description jc-alert-no-icon"
  data-show={true}
  style={undefined}
>
  <span
    className="jc-alert-message"
  >
    Error Text
  </span>
  <span
    className="jc-alert-description"
  >
    Error Description Error Description Error Description Error Description
  </span>
</div>
`;

exports[`<Alert /> Icon 图标 1`] = `
<div>
  <div
    className="jc-alert jc-alert-success"
    data-show={true}
    style={undefined}
  >
    <i
      className="jcicon jcicon-check-circle  jc-alert-icon"
      style={Object {}}
    />
    <span
      className="jc-alert-message"
    >
      Success Tips
    </span>
    <span
      className="jc-alert-description"
    />
  </div>
  <div
    className="jc-alert jc-alert-info"
    data-show={true}
    style={undefined}
  >
    <i
      className="jcicon jcicon-info-circle  jc-alert-icon"
      style={Object {}}
    />
    <span
      className="jc-alert-message"
    >
      Informational Notes
    </span>
    <span
      className="jc-alert-description"
    />
  </div>
  <div
    className="jc-alert jc-alert-warning"
    data-show={true}
    style={undefined}
  >
    <i
      className="jcicon jcicon-exclamation-circle  jc-alert-icon"
      style={Object {}}
    />
    <span
      className="jc-alert-message"
    >
      Warning
    </span>
    <span
      className="jc-alert-description"
    />
  </div>
  <div
    className="jc-alert jc-alert-error"
    data-show={true}
    style={undefined}
  >
    <i
      className="jcicon jcicon-cross-circle  jc-alert-icon"
      style={Object {}}
    />
    <span
      className="jc-alert-message"
    >
      Error
    </span>
    <span
      className="jc-alert-description"
    />
  </div>
  <div
    className="jc-alert jc-alert-success jc-alert-with-description"
    data-show={true}
    style={undefined}
  >
    <i
      className="jcicon jcicon-check-circle-o  jc-alert-icon"
      style={Object {}}
    />
    <span
      className="jc-alert-message"
    >
      Success Tips
    </span>
    <span
      className="jc-alert-description"
    >
      Detailed description and advices about successful copywriting.
    </span>
  </div>
  <div
    className="jc-alert jc-alert-info jc-alert-with-description"
    data-show={true}
    style={undefined}
  >
    <i
      className="jcicon jcicon-info-circle-o  jc-alert-icon"
      style={Object {}}
    />
    <span
      className="jc-alert-message"
    >
      Informational Notes
    </span>
    <span
      className="jc-alert-description"
    >
      Additional description and informations about copywriting.
    </span>
  </div>
  <div
    className="jc-alert jc-alert-warning jc-alert-with-description"
    data-show={true}
    style={undefined}
  >
    <i
      className="jcicon jcicon-exclamation-circle-o  jc-alert-icon"
      style={Object {}}
    />
    <span
      className="jc-alert-message"
    >
      Warning
    </span>
    <span
      className="jc-alert-description"
    >
      This is a warning notice about copywriting.
    </span>
  </div>
  <div
    className="jc-alert jc-alert-error jc-alert-with-description"
    data-show={true}
    style={undefined}
  >
    <i
      className="jcicon jcicon-cross-circle-o  jc-alert-icon"
      style={Object {}}
    />
    <span
      className="jc-alert-message"
    >
      Error
    </span>
    <span
      className="jc-alert-description"
    >
      This is an error message about copywriting.
    </span>
  </div>
</div>
`;

exports[`<Alert /> Style 四种样式 1`] = `
<div>
  <div
    className="jc-alert jc-alert-success jc-alert-no-icon"
    data-show={true}
    style={undefined}
  >
    <span
      className="jc-alert-message"
    >
      Success Text
    </span>
    <span
      className="jc-alert-description"
    />
  </div>
  <div
    className="jc-alert jc-alert-info jc-alert-no-icon"
    data-show={true}
    style={undefined}
  >
    <span
      className="jc-alert-message"
    >
      Info Text
    </span>
    <span
      className="jc-alert-description"
    />
  </div>
  <div
    className="jc-alert jc-alert-warning jc-alert-no-icon"
    data-show={true}
    style={undefined}
  >
    <span
      className="jc-alert-message"
    >
      Warning Text
    </span>
    <span
      className="jc-alert-description"
    />
  </div>
  <div
    className="jc-alert jc-alert-error jc-alert-no-icon"
    data-show={true}
    style={undefined}
  >
    <span
      className="jc-alert-message"
    >
      Error Text
    </span>
    <span
      className="jc-alert-description"
    />
  </div>
</div>
`;
